controllers/domachine_controller.go

```diff
@@ -225,11 +225,9 @@ func (r *DOMachineReconciler) reconcile(ctx context.Context, machineScope *scope
 	if droplet == nil {
 		droplet, err = computesvc.CreateDroplet(machineScope)
 		if err != nil {
-			errs := errors.Errorf("Failed to create droplet instance for DOMachine %s/%s: %v", domachine.Namespace, domachine.Name, err)
-			machineScope.SetFailureReason(capierrors.CreateMachineError)
-			machineScope.SetFailureMessage(errs)
-			r.Recorder.Event(domachine, corev1.EventTypeWarning, "InstanceCreatingError", errs.Error())
-			return reconcile.Result{}, errs
+			err = errors.Errorf("Failed to create droplet instance for DOMachine %s/%s: %v", domachine.Namespace, domachine.Name, err)
+			r.Recorder.Event(domachine, corev1.EventTypeWarning, "InstanceCreatingError", err.Error())
```

Comment:
for better experience we need to set the instance status here. to inform the user if there is an error.

```suggestion
			r.Recorder.Event(domachine, corev1.EventTypeWarning, "InstanceCreatingError", err.Error())
			machineScope.SetInstanceStatus(infrav1.DOResourceStatusErrored)
```

Reply:
Good catch -- updated and squashed.